### DISCUSSION TITLE
Is it possible to use the language server part as a separate app ?

### DISCUSSION THREAD

**Discussion by gayanper** [streetsidesoftware/vscode-spell-checker#2040] [Q&A]
I'm currently thinking of using the language server part of this plugin to integrate into Eclipse. It is possible as a improvement to provide language server component as a separate downloadable artifact ?

Another option is that I build it my self using source code and package it inside my Eclipse plugin, Not sure how the license will work out since I normally release my plugins under EPL.

Do you have any good recommendation on this to move forward ?

**Comment by Jason3S** [comment]
@gayanper,

Wonderful. A similar approach would be needed for: [Visual Studio extension · Issue #2617 · streetsidesoftware/cspell](https://github.com/streetsidesoftware/cspell/issues/2617)

Did you plan on using the VS Code language-sever protocol? It might be too tightly coupled to VS Code. I think it might be worth it to have a server implementation that is not tightly coupled to VS Code.

> Another option is that I build it my self using source code and package it inside my Eclipse plugin, Not sure how the license will work out since I normally release my plugins under EPL.

I think the server itself can have a different license, most likely MIT. But, any package that tightly bundles dictionaries, must be compatible with the licenses of the dictionaries. The VS Code extension uses a GPL license to handle the case where bundled (zipped with the extension) dictionaries carry the GPL license.
